src/App.tsx

```diff
@@ -1,10 +1,68 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+
 import { UserWarning } from './UserWarning';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { Main } from './components/Main/Main';
+import { Footer } from './components/Footer/Footer';
+import { Header } from './components/Header/Header';
+import { ErrorMessage } from './components/ErrorMessage/ErrorMessage';
+import { Errors } from './types/Errors';
+import { Filters } from './types/Filters';
 
-const USER_ID = 0;
+const USER_ID = 11338;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<Errors | ''>('');
+  const [errorVisibility, setErrorVisibility] = useState(false);
+  const [filter, setFilter] = useState(Filters.All);
+
+  const visibleTodos = useMemo(() => {
+    switch (filter) {
+      case Filters.Active: {
+        return [...todos].filter(todo => !todo.completed);
```

Comment:
```suggestion
        return todos.filter(todo => !todo.completed);
```
filter create the new array so we don't need to make copy here